gnomad/utils/transcript_annotation.py

```diff
@@ -416,10 +489,152 @@ def perform_tx_annotation_pipeline(
+    """
+    Create transcript annotation by region for loading into the gnomAD browser.
+
+    This function processes a Hail Table to create transcript annotations by region.
+    It calculates the mean expression proportion, handles missing values, and organizes
+    the data by genomic regions. The key fields are 'gene_id', 'exp_prop_mean', and
+    'tissues', ensuring regions split correctly based on changes in these fields.
+
+    .. table:: Input Hail Table
+        :widths: auto
+
+        +----------+---------+---------------+---------+---------+
+        | locus    | gene_id | exp_prop_mean | tissue1 | tissue2 |
+        +==========+=========+===============+=========+=========+
+        | chr1:1   | gene1   | 0.5           | 0.2     | 0.3     |
+        +----------+---------+---------------+---------+---------+
+        | chr1:2   | gene1   | 0.5           | 0.2     | 0.3     |
+        +----------+---------+---------------+---------+---------+
+        | chr1:3   | gene1   | 0.6           | 0.3     | 0.4     |
+        +----------+---------+---------------+---------+---------+
+        | chr1:4   | gene2   | 0.7           | 0.5     | 0.6     |
+        +----------+---------+---------------+---------+---------+
+        | chr1:5   | gene2   | 0.7           | 0.5     | 0.6     |
+        +----------+---------+---------------+---------+---------+
+        | chr1:6   | gene2   | 0.8           | 0.6     | 0.7     |
+        +----------+---------+---------------+---------+---------+
+
+    .. table:: Output Hail Table
+        :widths: auto
+
+        +---------+--------------------------------------------------------------+
+        | gene_id | regions                                                      |
+        +=========+==============================================================+
+        | gene1   | [{'chrom': 'chr1', 'start': 1, 'stop': 2, 'mean': 0.5,       |
+        |         | 'tissues': {'tissue1': 0.2, 'tissue2': 0.3}},                |
+        |         | {'chrom': 'chr1', 'start': 3, 'stop': 3, 'mean': 0.6,        |
+        |         | 'tissues': {'tissue1': 0.3, 'tissue2': 0.4}}]                |
+        +---------+--------------------------------------------------------------+
+        | gene2   | [{'chrom': 'chr1', 'start': 4, 'stop': 5, 'mean': 0.7,       |
+        |         | 'tissues': {'tissue1': 0.5, 'tissue2': 0.6}},                |
+        |         | {'chrom': 'chr1', 'start': 6, 'stop': 6, 'mean': 0.8,        |
+        |         | 'tissues': {'tissue1': 0.6, 'tissue2': 0.7}}                 |
+        +---------+--------------------------------------------------------------+
+
+    :param ht: Input Hail Table with transcript expression information.
+    :return: Hail Table with transcript annotations by region.
+    """
+    # Get a list of tissues and drop the 'tissues' field from the Table.
+    tissues = hl.eval(ht.tissues)
+    ht = ht.drop("tissues")
```

Comment:
do you need any of the other globals as well? couldn't you also drop `grange`, `vep_help`, `vep_config`?

Reply:
yeah, I guess those are really just because they are on the context table, whereas tissues was an expected global on the input ht, so that is all I was thinking about. I will switch to a `.select_globals()`